### PR TITLE
SOC-1016 Add tracking to Facebook connect / registration

### PR DESCRIPTION
**Pull request by rafalkalinski**
https://wikia-inc.atlassian.net/browse/SOC-1016

This branch contains code from SOC-490 and SOC-903. So it should be merged after those two.

### CODE REVIEW

File: front/scripts/auth/facebook/FacebookRegistration.ts [hunk 1]
```diff
@@ -130,6 +132,7 @@ class FacebookRegistration {
 		};
 
 		facebookRegistrationXhr.onerror = (e: Event) => {
+			this.track('facebook-login-registration', Mercury.Utils.trackActions.error);
```
Comment (kvas-damian) [inline front/scripts/auth/facebook/FacebookRegistration.ts]: Shouldn't label be ```facebook-login-registration-helios``` ?

File: front/scripts/main/controllers/ApplicationController.ts [hunk 2]
```diff
@@ -7,12 +7,15 @@ App.ApplicationController = Em.Controller.extend(App.LoadingSpinnerMixin, App.Al
 	// This has to be here because we need to access media from ArticleController model to open lightbox
 	// TODO: Should be refactored when decoupling article from application
 	needs: ['article'],
-	queryParams: ['file', 'map', {
-		noAds: 'noads'
-	}],
+	queryParams: ['file', 'map',
```
Comment (kvas-damian) [inline front/scripts/main/controllers/ApplicationController.ts]: What is this change for?

Comment (rogatty) [inline front/scripts/main/controllers/ApplicationController.ts]: My bet is that SOC-903 is behind dev further than this branch.

File: front/scripts/auth/facebook/FacebookLogin.ts [hunk 3]
```diff
@@ -81,8 +82,8 @@ class FacebookLogin {
 
 		facebookTokenXhr.onload = (e: Event): void => {
 			var status: number = (<XMLHttpRequest> e.target).status;
-
 			if (status === HttpCodes.OK) {
+				this.track('facebook-login-helios', Mercury.Utils.trackActions.success);
```
Comment (bkoval) [inline front/scripts/auth/facebook/FacebookLogin.ts]: Maybe adding 'succesful' to the label would be more descriptive?

File: front/scripts/auth/facebook/FacebookRegistration.ts [hunk 4]
```diff
@@ -138,4 +141,13 @@ class FacebookRegistration {
 		facebookRegistrationXhr.setRequestHeader('Content-type', 'application/x-www-form-urlencoded');
 		facebookRegistrationXhr.send(this.urlHelper.urlEncode(data));
 	}
+
+	private track (label: string, action: string): void {
+		M.track({
+			trackingMethod: 'both',
+			action: action,
+			category: 'user-sign-up-mobile',
```
Comment (bkoval) [inline front/scripts/auth/facebook/FacebookRegistration.ts]: Are you sure there should be a dash between "sign" and "up"? :santa:

**Comment by kvas-damian** [comment]
@rafalkalinski did you consider tracking different error types returned by helios?

File: front/scripts/auth/facebook/FacebookRegistration.ts (hunk 1 above)
Comment (rafalkalinski) [inline front/scripts/auth/facebook/FacebookRegistration.ts]: changed to 'facebook-signup-join-wikia-error' due ticket comment...

File: front/scripts/main/controllers/ApplicationController.ts (hunk 2 above)
Comment (rafalkalinski) [inline front/scripts/main/controllers/ApplicationController.ts]: That's right and SOC-903 (and the diff) is fixed by now.

File: front/scripts/auth/facebook/FacebookLogin.ts (hunk 3 above)
Comment (rafalkalinski) [inline front/scripts/auth/facebook/FacebookLogin.ts]: added '-success' and '-error' postfixes

File: front/scripts/auth/facebook/FacebookRegistration.ts (hunk 4 above)
Comment (rafalkalinski) [inline front/scripts/auth/facebook/FacebookRegistration.ts]: I was following the ticket description. 
Fixed.

**Comment by rafalkalinski** [comment]
@kvas-damian that's a subject of another ticket (SOC-478)

**Comment by kvas-damian** [comment]
LGTM

**Comment by bkoval** [comment]
LGTM :sandal:

**Pull request merged**